pkg/controller/utils/utils.go

```diff
@@ -21,6 +18,15 @@ import (
 	routev1 "github.com/openshift/api/route/v1"
 )
 
+// DockerConfigJSON type would be be used to unmarshal pull secrets
+type DockerConfigJSON struct {
+	Auths       DockerConfig      `json:"auths"`
+	HTTPHeaders map[string]string `json:"HttpHeaders,omitempty"`
```

Comment:
Are we assuming that the HTTP headers are irrelevant and so do not need to be merged?

Reply:
Right. `HTTPHeaders` not relevent

Reply:
Then let's remove `HTTPHeaders` from the struct. I would rather remove the headers entirely instead of having undocumented behavior where the headers are preserved from the global pull secret and ignored from the local pull secret.